cirq-core/cirq/ops/pauli_string_phasor.py

```diff
@@ -58,8 +59,119 @@ def __init__(
                 in the form of the t in (-1)**t = exp(i pi t).
 
         Raises:
-            ValueError: If the given pauli string does not have eignevalues +1
-                or -1.
+            ValueError: If coefficient is not 1 or -1.
+        """
+        gate = PauliStringPhasorGate(
+            dps.DensePauliString(pauli_string.values(), coefficient=pauli_string.coefficient),
+            exponent_neg=exponent_neg,
+            exponent_pos=exponent_pos,
+        )
+        super().__init__(gate, tuple(pauli_string))
+        self._pauli_gate: 'PauliStringPhasorGate' = gate
+
+    @property
+    def exponent_neg(self):
+        return self._pauli_gate.exponent_neg
+
+    @property
+    def exponent_pos(self):
+        return self._pauli_gate.exponent_pos
+
+    @property
+    def pauli_string(self):
+        return self._pauli_gate.pauli_string.on(*self.qubits)
```

Comment:
Why do this rather expensive operation instead of directly storing the created `pauli_string` in this class at construction? To keep things consistent?